app/src/js/components/Providers/overview.js

```diff
@@ -14,111 +14,95 @@ import {
-  }
+  const providers = useSelector((state) => state.providers);
+  const stats = useSelector((state) => state.stats);
 
-  queryStats() {
-    this.props.dispatch(
-      getCount({
-        type: 'collections',
-        field: 'providers',
-      })
-    );
-  }
-
-  generateQuery() {
-    const { queryParams } = this.props;
-    return { ...queryParams };
-  }
+  useEffect(() => {
+    dispatch(getCount({
+      type: 'collections',
+      field: 'providers',
+    }));
+  }, [dispatch]);
 
-  render() {
-    const { providers, stats } = this.props;
-    const { list } = providers;
-    const { count, queriedAt } = list.meta;
+  const generateQuery = () => ({ ...queryParams });
 
-    const bulkActions = [
-      {
-        Component: (
-          <Link
-            className="button button--green button--add button--small form-group__element"
-            to={(location) => ({
-              pathname: '/providers/add',
-              search: getPersistentQueryParams(location),
-            })}
-          >
-            Add Provider
-          </Link>
-        ),
-      },
-    ];
+  const { list } = providers;
+  const { count, queriedAt } = list.meta;
 
-    // Incorporate the collection counts into the `list`
-    const mutableList = cloneDeep(list);
-    const collectionCounts = get(stats.count, 'data.collections.count', []);
+  const mutableList = cloneDeep(list);
+  const collectionCounts = get(stats.count, 'data.collections.count', []);
 
-    mutableList.data.forEach((d) => {
-      d.collections = get(
-        collectionCounts.find((c) => c.key === d.name),
-        'count',
-        0
-      );
-    });
-    return (
-      <div className="page__component">
-        <Helmet>
-          <title> Provider </title>
-        </Helmet>
-        <section className="page__section page__section__header-wrapper">
-          <h1 className="heading--large heading--shared-content with-description">
-            Provider Overview
-          </h1>
-          {lastUpdated(queriedAt)}
-        </section>
-        <section className="page__section">
-          <div className="heading__wrapper--border">
-            <h2 className="heading--medium heading--shared-content">
-              Ingesting Providers
-              <span className="num-title">{count ? `${count}` : 0}</span>
-            </h2>
-          </div>
-          <List
-            list={mutableList}
-            action={listProviders}
-            tableColumns={tableColumns}
-            query={this.generateQuery()}
-            bulkActions={bulkActions}
-            rowId="name"
-            initialSortId="updatedAt"
-            filterAction={filterProviders}
-            filterClear={clearProvidersFilter}
-            tableId="providers"
-          ></List>
-        </section>
-      </div>
+  mutableList.data.forEach((d) => {
+    d.collections = get(
+      collectionCounts.find((c) => c.key === d.name),
+      'count',
+      0
     );
-  }
-}
+  });
+
+  const bulkActions = [
```

Comment:
Same question as above. Is anything different in here? It looks like a straight-up move of a large block of code with no changes...

Reply:
I had to convert the component from class to functional for React Router v6. Can't add the hooks and functionality into the older class version. Constructors are no longer used. I put the code in order: hooks, state, useEffect, variables and functions about `list` and `data` and then the actions (`bulkActions`). The interactions didn't change.